grails-app/controllers/org/pih/warehouse/core/DocumentController.groovy

```diff
@@ -182,7 +182,8 @@ class DocumentController {
             flash.message = "${warehouse.message(code: 'document.documentTooLarge.message')}"
         } else if (file.size < 10 * 1024 * 1000) {
             log.info "Creating new document "
-            def typeId = command?.typeId ?: "0"
+            // Document type with id 9 is "Other" and it's default in case there's no document type chosen
+            def typeId = command?.typeId ?: 9
```

Comment:
Can you make a small change? All constants like this should be declared in the Constants file. And note that the ID has to be a String.
```
static final String DEFAULT_DOCUMENT_TYPE_ID = "9"
```
And then replace the 9 with the default value.
```
def typeId = command?.typeId ?: Constants.DEFAULT_DOCUMENT_TYPE_ID;
```

Reply:
Otherwise it looks good.